utils/navigate.py

```diff
@@ -0,0 +1,47 @@
+from utils import InstanceClassMethod
+from utils.log import logger
+
+
+class NavigateWrap(object):
+    def __init__(self, *args, **kwargs):
+        self.cls = args[0]
+
+    def __call__(self, f):
+        self.cls._nav_steps[f.__name__] = f
+        return f
+
+
+class Navigatable(object):
+    def __init__(self, *args, **kwargs):
+        self.navigate = self.instance_navigate
+
+    @InstanceClassMethod
+    def navigate(self, name):
```

Comment:
i think for clarity we might want to replace this one iwth a `ClassORInstanceMethod` that has the prototype `def navigate(cls_or_self, name):`  ´</nitpick>`